### PR TITLE
Introduce ClusterPolicy config-syncer-secret-generation-from-rancher-capi

### LINKED ISSUE #418: [Sample] Kubeops Config Syncer Secret Generation From Rancher CAPI Secret
### Problem Statement

The AppsCode Config Syncer supports syncing ConfigMaps/Secrets across Clusters. This feature requires a pre-existing kubeconfig Secret described [here:](https://appscode.com/products/kubed/v0.12.0/guides/config-syncer/inter-cluster/#deploy-kubed). The Helm chart for this tool has a property `config.kubeconfigContent` which is used to create the above secret if it is provided.

The proposed Kyverno Generate policy entitled `config-syncer-secret-generation-from-rancher-capi` is intended to automate the generation of this kubeconfig Secret on Rancher Management Clusters so that ConfigMaps/Secrets can easily be synced to downstream clusters.

### Solution Description

The metadata for the proposed ClusterPolicy is included below:

```
kind: ClusterPolicy
metadata:
  name: config-syncer-secret-generation-from-rancher-capi
  annotations:
    policies.kyverno.io/title: Kubeops Config Syncer Secret Generation From Rancher CAPI Secret
    policies.kyverno.io/category: Kubeops
    policies.kyverno.io/severity: medium
    policies.kyverno.io/subject: Secret
    kyverno.io/kyverno-version: 1.8.0
    policies.kyverno.io/minversion: 1.7.0
    kyverno.io/kubernetes-version: "1.23"
    policies.kyverno.io/description: >-
      This policy generates and synchronizes a Kubeops Config Syncer merged kubeconfig
      Secret from Rancher managed cluster CAPI secrets. This kubeconfig Secret is
      required by the Kubeops Config Syncer for it to sync ConfigMaps/Secrets from
      the Rancher management cluster to downstream clusters.
```

### Example "Good" Resource

_No response_

### Example "Bad" Resource

_No response_

### Other Comments

_No response_

### Slack discussion

_No response_

### Troubleshooting

- [X] I have read and followed the documentation AND the [troubleshooting guide](https://kyverno.io/docs/troubleshooting/).
- [X] I have verified the policy does not exist in the [samples library](https://kyverno.io/policies/).

### PR DESCRIPTION
**Pull request by ron1**
## Related Issue(s)

Closes #418

## Description
Implement ClusterPolicy config-syncer-secret-generation-from-rancher-capi.

Note that Kyverno 1.8.0 appears to contain a JMESPath bug that currently prevents this policy from working. Below, please find the contents of the Config Syncer/Kubed Secret generated by this policy. Specifically, notice that the `contexts` json element was rendered incorrectly. This may be due to the fact that the `contexts.context` element is complex. If I change `contexts.context` to be a simple property like `context: metadata.name`, the `contexts` json element renders correctly.

This issue is currently preventing me from using this ClusterPolicy. Please investigate.
```
$ kubectl -n kube-system get secret kubed -o jsonpath='{.data.kubeconfig}' | base64 -d
{
  "apiVersion": "v1",
  "kind": "Config",
  "clusters": [{"cluster":{"certificate-authority-data":"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJ2VENDQVdPZ0F3SUJBZ0lCQURBS0JnZ3Foa2pPUFFRREFqQkdNUnd3R2dZRFZRUUtFeE5rZVc1aGJXbGoKYkdsemRHVnVaWEl0YjNKbk1TWXdKQVlEVlFRRERCMWtlVzVoYldsamJHbHpkR1Z1WlhJdFkyRkFNVFkyTlRNNApOakV5T1RBZUZ3MHlNakV3TVRBd056RTFNamxhRncwek1qRXdNRGN3TnpFMU1qbGFNRVl4SERBYUJnTlZCQW9UCkUyUjVibUZ0YVdOc2FYTjBaVzVsY2kxdmNtY3hKakFrQmdOVkJBTU1IV1I1Ym1GdGFXTnNhWE4wWlc1bGNpMWoKWVVBeE5qWTFNemcyTVRJNU1Ga3dFd1lIS29aSXpqMENBUVlJS29aSXpqMERBUWNEUWdBRStwenNjNVhMdWttSQp6aVNLeThkZ0FiTnB1NFFWR0VnSVBzTGwyN3JsVTM0SHh5TmhxZEtNN05rcm1ZWHkrTTZIc2dxR3NiSWpiZjYxCk9kZm94b1RwRGFOQ01FQXdEZ1lEVlIwUEFRSC9CQVFEQWdLa01BOEdBMVVkRXdFQi93UUZNQU1CQWY4d0hRWUQKVlIwT0JCWUVGRHBseTZON2hoNUdybDdhKzA4TXY0UDhpWXFPTUFvR0NDcUdTTTQ5QkFNQ0EwZ0FNRVVDSUhidwo0N0hyZmx6aEVWbTBUZXpraFV5QjI4SWJWTkJPbmd5NVFOWDk3SXpiQWlFQTY2VUtiSC92WWIyekYyZUhKV2JCCmtGaFdkQW1rU0NmZW5VNnk1UkVRbHlVPQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0t","server":"https://<my-server>.<my-domain>/k8s/clusters/c-m-74lz8h8w"},"name":"k8sds-1-kubeconfig"},{"cluster":{"certificate-authority-data":"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJ2VENDQVdPZ0F3SUJBZ0lCQURBS0JnZ3Foa2pPUFFRREFqQkdNUnd3R2dZRFZRUUtFeE5rZVc1aGJXbGoKYkdsemRHVnVaWEl0YjNKbk1TWXdKQVlEVlFRRERCMWtlVzVoYldsamJHbHpkR1Z1WlhJdFkyRkFNVFkyTlRNNApOakV5T1RBZUZ3MHlNakV3TVRBd056RTFNamxhRncwek1qRXdNRGN3TnpFMU1qbGFNRVl4SERBYUJnTlZCQW9UCkUyUjVibUZ0YVdOc2FYTjBaVzVsY2kxdmNtY3hKakFrQmdOVkJBTU1IV1I1Ym1GdGFXTnNhWE4wWlc1bGNpMWoKWVVBeE5qWTFNemcyTVRJNU1Ga3dFd1lIS29aSXpqMENBUVlJS29aSXpqMERBUWNEUWdBRStwenNjNVhMdWttSQp6aVNLeThkZ0FiTnB1NFFWR0VnSVBzTGwyN3JsVTM0SHh5TmhxZEtNN05rcm1ZWHkrTTZIc2dxR3NiSWpiZjYxCk9kZm94b1RwRGFOQ01FQXdEZ1lEVlIwUEFRSC9CQVFEQWdLa01BOEdBMVVkRXdFQi93UUZNQU1CQWY4d0hRWUQKVlIwT0JCWUVGRHBseTZON2hoNUdybDdhKzA4TXY0UDhpWXFPTUFvR0NDcUdTTTQ5QkFNQ0EwZ0FNRVVDSUhidwo0N0hyZmx6aEVWbTBUZXpraFV5QjI4SWJWTkJPbmd5NVFOWDk3SXpiQWlFQTY2VUtiSC92WWIyekYyZUhKV2JCCmtGaFdkQW1rU0NmZW5VNnk1UkVRbHlVPQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0t","server":"https://<my-server>.<my-domain>/k8s/clusters/c-m-jdwpx2dh"},"name":"k8sds-2-kubeconfig"}]
,
  "users": [{"name":"k8sds-1-kubeconfig","user":{"token":"u-bdrxdhr4cm:kgc6txcgspr2mjsj2f2s5nfgfwbnprl65bwcv4cgxvmxmk5sfgntqp"}},{"name":"k8sds-2-kubeconfig","user":{"token":"u-zt4s4t6rpu:v5spt8mzsscg5vtq278gjtv6dc5j56lzc6g5cmxj7fjnrrctvjblvw"}}]
,
  "contexts": {{ secretList | parse_yaml(@) | items[?metadata.name.contains(@, '-kubeconfig')].{
    "name": metadata.name,
    "context": { "cluster": metadata.name, "user": metadata.name } }
}}

}
```

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.

### CODE REVIEW

**Comment by ron1** [comment]
I force pushed an improvement to the context secretList apiCall as requested. Unfortunately, the improvement did not fix the bug. I also tried replacing all references to `context` and `contexts` in my variable names with a different value. This also had no impact on the result.

**Comment by chipzoller** [comment]
DCO sign-off is required here.

File: kubeops/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Kubeops Policy Examples
+
+This directory contains examples in order to apply policies on
+[Kubeops](https://github.com/kubeops) curtated tools.
```
Comment (chipzoller) [inline kubeops/README.md:4]: "curated"?

**Review by chipzoller — APPROVED** [review]
Cool!

**Pull request merged**